### PR TITLE
Refactoring - move setting password to connectSensors

### PR DESCRIPTION
**Pull request by bostko**
- set password from config

@duncangrant @iyovcheva can you review it?

### CODE REVIEW

File: ambari/src/main/java/io/brooklyn/ambari/server/AmbariServerImpl.java [hunk 1]
```diff
@@ -151,11 +125,10 @@ protected void connectSensors() {
                 .period(500, TimeUnit.MILLISECONDS)
                 .baseUri(ambariUri)
                 .poll(new HttpPollConfig<Boolean>(URL_REACHABLE)
-                        .onSuccess(HttpValueFunctions.responseCodeEquals(200))
```
Comment (bostko) [inline ambari/src/main/java/io/brooklyn/ambari/server/AmbariServerImpl.java]: HttpFeed.successChecker is used for success check.

**Comment by bostko** [comment]
Do not merge, testing.

**Comment by bostko** [comment]
@duncangrant I tested it and it works fine with rebinding.

File: launcher/src/main/assembly/files/conf/logback.xml [hunk 2]
```diff
@@ -24,7 +24,8 @@
        this includes as described in the Logging section of the brooklyn users guide.
        this project already supplies a logback-custom.xml which gets included to log our
        classes at debug level and write to a log file named after this project -->
-
+  <property name="logging.basename" scope="context" value="brooklyn-ambari" />
```
Comment (duncangrant) [inline launcher/src/main/assembly/files/conf/logback.xml]: This is probably a good idea but should be in a separate PR

File: pom.xml [hunk 3]
```diff
@@ -48,7 +48,7 @@
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
         <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
         <project.entry>io.brooklyn.ambari.AmbariMain</project.entry>
-        <brooklyn.version>0.9.0-SNAPSHOT</brooklyn.version> <!-- BROOKLYN_VERSION -->
+        <brooklyn.version>0.9.0</brooklyn.version> <!-- BROOKLYN_VERSION -->
```
Comment (duncangrant) [inline pom.xml:51]: Should be a separate PR

File: ambari/src/main/java/io/brooklyn/ambari/server/AmbariServerImpl.java [hunk 4]
```diff
@@ -471,4 +448,41 @@ public void setFqdn(String fqdn) {
...
+        generatePassword();
+    }
+
+    /**
+     * Need to wait server to start before resetting the password.
+     */
+    protected void generatePassword() {
+        if (Strings.isBlank(getAttribute(AmbariServer.PASSWORD))) {
+            String password;
+            if (Strings.isBlank(getConfig(AmbariServer.PASSWORD))) {
```
Comment (duncangrant) [inline ambari/src/main/java/io/brooklyn/ambari/server/AmbariServerImpl.java]: I'd change so that
String password = getConfig(AmbariServer.PASSWORD)
That way there is only a single call to getConfig

File: ambari/src/main/java/io/brooklyn/ambari/server/AmbariServer.java [hunk 5]
```diff
@@ -58,7 +58,7 @@
     AttributeSensor<String> CLUSTER_STATE = Sensors.newStringSensor("ambari.server.clusterState");
 
     BasicAttributeSensorAndConfigKey.StringAttributeSensorAndConfigKey USERNAME =
-            new BasicAttributeSensorAndConfigKey.StringAttributeSensorAndConfigKey("ambari.server.http.user", "Ambari Server's http username", "admin");
+            new BasicAttributeSensorAndConfigKey.StringAttributeSensorAndConfigKey("ambari.server.http.user", "Ambari Server's http username. It is always admin", "admin");
```
Comment (duncangrant) [inline ambari/src/main/java/io/brooklyn/ambari/server/AmbariServer.java]: Possibly should read "... default is admin".  As it currently sounds like it won't change.

File: ambari/src/main/java/io/brooklyn/ambari/server/AmbariServerImpl.java [hunk 6]
```diff
@@ -471,4 +448,41 @@ public void setFqdn(String fqdn) {
...
+                password = Identifiers.makeRandomId(10);
```
Comment (duncangrant) [inline ambari/src/main/java/io/brooklyn/ambari/server/AmbariServerImpl.java]: I think we've agreed that this isn't a good way to generate a password.  If password isn't set then I think this should be an error.  That way we still have a default of ambari but if someone deliberately sets it to blank it could fail with a sensible error message.  If someone wants a generated password then brooklyn has a CreatePasswordSensor.

File: ambari/src/main/java/io/brooklyn/ambari/server/AmbariServerImpl.java [hunk 7]
```diff
@@ -91,8 +91,7 @@
 
     private RestAdapter restAdapter;
 
-    //TODO clearly needs changed
-    private UsernamePasswordCredentials usernamePasswordCredentials = getUsernamePasswordCredentials();
+    private UsernamePasswordCredentials usernamePasswordCredentials;
 
     private static final String USERNAME = "admin";
     private static final String DEFAULT_PASSWORD = "admin";
```
Comment (duncangrant) [inline ambari/src/main/java/io/brooklyn/ambari/server/AmbariServerImpl.java]: Not default any more so maybe rename as INITIAL_PASSWORD?

File: ambari/src/main/java/io/brooklyn/ambari/server/AmbariServer.java [hunk 8]
```diff
@@ -58,7 +58,7 @@
...
 
     BasicAttributeSensorAndConfigKey.StringAttributeSensorAndConfigKey PASSWORD =
             new BasicAttributeSensorAndConfigKey.StringAttributeSensorAndConfigKey("ambari.server.http.password", "Ambari Server's http password");
```
Comment (duncangrant) [inline ambari/src/main/java/io/brooklyn/ambari/server/AmbariServer.java:63]: Should default to admin so this doesn't change expected behaviour.

File: ambari/src/main/java/io/brooklyn/ambari/server/AmbariServer.java [hunk 9]
```diff
@@ -58,7 +58,7 @@
     AttributeSensor<String> CLUSTER_STATE = Sensors.newStringSensor("ambari.server.clusterState");
 
     BasicAttributeSensorAndConfigKey.StringAttributeSensorAndConfigKey USERNAME =
-            new BasicAttributeSensorAndConfigKey.StringAttributeSensorAndConfigKey("ambari.server.http.user", "Ambari Server's http username", "admin");
+            new BasicAttributeSensorAndConfigKey.StringAttributeSensorAndConfigKey("ambari.server.http.user", "Ambari Server's http username. It is always admin", "admin");
```
Comment (duncangrant) [inline ambari/src/main/java/io/brooklyn/ambari/server/AmbariServer.java:61]: Seems daft to have a config key that has no effect when it is set - should we just delete this?

**Comment by duncangrant** [comment]
Happy with this.  Couple of comments that are possibly worth doing.

Can you create separate PRs (or at least commits) for config changes and version update.

File: ambari/src/main/java/io/brooklyn/ambari/server/AmbariServer.java (hunk 9 above)
Comment (ivanayov) [inline ambari/src/main/java/io/brooklyn/ambari/server/AmbariServer.java:61]: This would be helpful if we want to be able in future to configure new users. It's fast and easy to implement as it's very similar to the `updateUser` implementation, but I'm not sure if we want it. WDYT, is it a reason to keep this code?

**Comment by ivanayov** [comment]
Thanks @bostko !

**Comment by bostko** [comment]
Testing....

**Comment by bostko** [comment]
@duncangrant Tested and it works ok.

File: ambari/src/main/java/io/brooklyn/ambari/server/AmbariServerImpl.java [hunk 10]
```diff
@@ -471,4 +448,40 @@ public void setFqdn(String fqdn) {
...
+    /**
+     * Need to wait server to start before resetting the password.
+     */
+    protected void generatePassword() {
+        String password = getAttribute(AmbariServer.PASSWORD);
+        if (password == null) {
+            password = getConfig(AmbariServer.PASSWORD);
+            if (password == null) {
+                password = Identifiers.makeRandomId(10);
+            }
```
Comment (duncangrant) [inline ambari/src/main/java/io/brooklyn/ambari/server/AmbariServerImpl.java:467]: Could you use ternary operator here?
